discovery-infra/tests/test_logs.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import tarfile
+import time
+
+from assisted_service_client.rest import ApiException
+from tests.base_test import BaseTest
+
+class TestDownloadLogs(BaseTest):
+    def setup_hosts(self, cluster_id, api_client, node_controller):
+        '''setup nodes from ISO image and wait until they are registered'''
+        # Generate and download cluster ISO
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        # Boot nodes into ISO
+        node_controller.start_all_nodes()
+        # Wait until hosts are discovered and update host roles
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client)
+    
+    def prepare_for_installation(self, cluster_id, api_client, node_controller):
+        '''set roles and network params to prepare the cluster for install'''
+        self.set_host_roles(cluster_id=cluster_id, api_client=api_client)
+        self.set_network_params(cluster_id=cluster_id,
+                                 api_client=api_client,
+                                 controller=node_controller
+                                 )
+
+    def install_cluster_and_wait(self, cluster_id, api_client):
+        '''start the cluster and wait for host to go to installing-in-progress state'''
+        # Start cluster install
+        self.start_cluster_install(cluster_id=cluster_id, api_client=api_client)
+        # wait until all nodes are in Installed status, and the cluster moved to status installed
```

Comment:
Please add self.wait_for_nodes_to_install(cluster_id=cluster_id, api_client=api_client), so if one of the nodes fails to install, then test fails rather then wait for timeout